src/targets/gpu/fuse_mlir.cpp

```diff
@@ -678,93 +680,140 @@ using find_mlir_standalone_dot_op         = find_mlir_standalone_op<&is_mlir_dot
 
 struct find_mlir_standalone_attention_op
 {
+    mlir_mode dot_mode = mlir_mode::none;
+
     auto matcher() const
     {
-        return match::name("gpu::pre_gemm_softmax_gemm").bind("gemm_softmax_gemm");
+        auto gemm1 =
+            match::skip(match::name("contiguous"))(match::used_once(), is_mlir_dot(dot_mode))
+                .bind("gemm1");
+        auto fused_reduce =
+            match::name("fused_reduce")(match::used_once(), match::any_of[match::inputs()](gemm1))
+                .bind("fused_reduce");
+        return is_mlir_dot(dot_mode)(match::arg(0)(fused_reduce)).bind("gemm2");
     }
 
-    void apply(module_pass_manager& mpm, const match::matcher_result& r) const
+    std::unordered_map<instruction_ref, instruction_ref>
+    invert_map_ins(const std::unordered_map<instruction_ref, instruction_ref>& map_ins) const
     {
-        static size_t counter  = 0;
-        module_ref mm          = mpm.create_module("mlir_" + std::to_string(counter++));
-        auto gemm_softmax_gemm = r.instructions["gemm_softmax_gemm"];
-        mm->set_bypass();
+        std::unordered_map<instruction_ref, instruction_ref> inverse_map;
+        for(auto const& [i1, i2] : map_ins)
```

Comment:
Can you name the variables like `key, value`? I think it will make it a little clearer.